### PR TITLE
Bump rs-soroban-env

### PR DESCRIPTION
**Pull request by sisuresh**
# Description

# Checklist
- [ ] Reviewed the [contributing](https://github.com/stellar/stellar-core/blob/master/CONTRIBUTING.md#submitting-changes) document
- [ ] Rebased on top of master (no merge commits)
- [ ] Ran `clang-format` v8.0.0 (via `make format` or the Visual Studio extension)
- [ ] Compiles
- [ ] Ran all tests
- [ ] If change impacts performance, include supporting evidence per the [performance document](https://github.com/stellar/stellar-core/blob/master/performance-eval/performance-eval.md)

### CODE REVIEW

File: src/rust/src/contract.rs [hunk 1]
```diff
@@ -378,9 +378,11 @@ fn storage_footprint_to_ledger_footprint(
             AccessType::ReadWrite => read_write.push(k.clone()),
         }
     }
+    let read_only_not_boxed: Vec<LedgerKey> = read_only.into_iter().map(|x| *x).collect();
```
Comment (sisuresh) [inline src/rust/src/contract.rs:381]: @graydon do you know of a better way to go from `Vec<Box<LedgerKey>>` to `VecM<LedgerKey>`?

Comment (leighmcculloch) [reply src/rust/src/contract.rs:381]: I don't see significantly better way of doing this. If you only needed references to the ledger keys you could get a Vec<&LedgerKey>, but you actually need owned values for the LedgerFootprint. If we're going to be doing this type of conversion frequently we could add a `TryFrom<Vec<Box<T>>> for VecM<T>` to `stellar_xdr`.

Comment (graydon) [reply src/rust/src/contract.rs:381]: Yeah I think this is the best bet for now.

**Review by graydon — APPROVED** [review]
LGTM! Glad it was so small a set of changes, thanks.

**Comment by sisuresh** [comment]
I just realized the wasmi versions between core and env didn't match up. Updated that as well in another commit.

**Review by graydon — APPROVED** [review]
LGTM!

**Comment by graydon** [comment]
r+ c4ec885

**Pull request merged**